x-pack/plugins/security_solution/common/search_strategy/security_solution/network/tls.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IEsSearchResponse } from '../../../../../../../src/plugins/data/common';
+import { CursorType, Inspect, Maybe, PageInfoPaginated, RequestOptionsPaginated } from '..';
+
+export interface TlsBuckets {
+  key: string;
+  timestamp?: {
+    value: number;
+    value_as_string: string;
+  };
+  subjects: {
+    buckets: Readonly<Array<{ key: string; doc_count: number }>>;
+  };
+  ja3: {
+    buckets: Readonly<Array<{ key: string; doc_count: number }>>;
+  };
+  issuers: {
+    buckets: Readonly<Array<{ key: string; doc_count: number }>>;
+  };
+  not_after: {
+    buckets: Readonly<Array<{ key: number; key_as_string: string; doc_count: number }>>;
+  };
+}
+
+export interface TlsNode {
+  _id?: Maybe<string>;
+  timestamp?: Maybe<string>;
+  notAfter?: Maybe<string[]>;
+  subjects?: Maybe<string[]>;
+  ja3?: Maybe<string[]>;
+  issuers?: Maybe<string[]>;
+}
+
+export enum FlowTargetSourceDest {
+  destination = 'destination',
+  source = 'source',
+}
+
+export enum TlsFields {
+  _id = '_id',
+}
+
+export interface TlsEdges {
+  node: TlsNode;
+  cursor: CursorType;
+}
+
+export enum Direction {
+  asc = 'asc',
+  desc = 'desc',
+}
+
+export interface TlsSortField {
```

Comment:
ok I think that I would like us to refactor `SortField` to be like that _below_.  Therefore, now we can have a generic/strict type if need it and TlsSortField can be replaced by `SortField<TlsFields>`.
Does that make sense?
```
export interface SortField<Field = string> {
  field: Field;
  direction: Direction;
}
```